src/object/cloneDeep.ts

```diff
@@ -184,7 +185,9 @@ function cloneDeepImpl<T>(obj: T, stack = new Map<any, any>()): T {
   }
 
   if (typeof obj === 'object' && obj !== null) {
-    const result = {};
+    const result =
+      typeof obj.constructor === 'function' && !isPrototype(obj) ? Object.create(Object.getPrototypeOf(obj)) : {};
```

Comment:
I wonder why we check for whether `obj.constructor` is `function` and `obj` is not a prototype here -- isn't just `Object.create(Object.getPrototypeOf(obj))` enough?

Reply:
I assumed that calling `Object.create()` and `Object.getPrototypeOf()` on every object would be expensive and hurt performance, but it's actually the opposite! Using just the `Object.create(Object.getPrototypeOf(obj))` improved performance by about 15%. I updated the PR.